components/host-dashboard/AddFundsModal.js

```diff
@@ -0,0 +1,302 @@
+);
+
+AmountDetailsLine.propTypes = {
+  label: PropTypes.node,
+  currency: PropTypes.string.isRequired,
+  value: PropTypes.number,
+  isLargeAmount: PropTypes.bool,
+};
+
+const addFundsMutation = gqlV2/* GraphQL */ `
+  mutation AddFunds(
+    $fromAccount: AccountReferenceInput!
+    $account: AccountReferenceInput!
+    $amount: AmountInput!
+    $description: String!
+    $hostFeePercent: Int!
+    $platformFeePercent: Int
+  ) {
+    addFunds(
+      account: $account
+      fromAccount: $fromAccount
+      amount: $amount
+      description: $description
+      hostFeePercent: $hostFeePercent
+      platformFeePercent: $platformFeePercent
+    ) {
+      id
+      toAccount {
+        id
+        stats {
+          balance {
+            valueInCents
+          }
+        }
+      }
+    }
+  }
+`;
+
+const getInitialValues = values => ({
+  amount: null,
+  hostFeePercent: null,
+  platformFeePercent: OC_FEE_PERCENT,
+  description: '',
+  fromAccount: null,
+  ...values,
+});
+
+const validate = values => {
+  return requireFields(values, ['amount', 'fromAccount', 'description']);
+};
+
+// Build an account reference. Compatible with accounts from V1 and V2.
+const buildAccountReference = input => {
+  return typeof input.id === 'string' ? { id: input.id } : { legacyId: input.legacyId };
+};
+
+const AddFunds = ({ host, collective, ...props }) => {
+  const { LoggedInUser } = useUser();
+  const [submitAddFunds, { error }] = useMutation(addFundsMutation, { context: API_V2_CONTEXT });
+  const defaultHostFeePercent = collective.hostFeePercent;
+
+  return (
+    <StyledModal width="100%" maxWidth={435} {...props}>
+      <CollectiveModalHeader collective={collective} />
+      <Formik
+        initialValues={getInitialValues({ hostFeePercent: defaultHostFeePercent, account: collective })}
+        validate={validate}
+        onSubmit={values =>
+          submitAddFunds({
+            variables: {
+              ...values,
+              amount: { valueInCents: values.amount },
+              fromAccount: buildAccountReference(values.fromAccount),
+              account: buildAccountReference(values.account),
+            },
+          }).then(props.onClose)
+        }
+      >
+        {({ values, isSubmitting, isValid, dirty }) => {
+          const hostFeePercent = isNaN(values.hostFeePercent) ? defaultHostFeePercent : values.hostFeePercent;
+          const platformFeePercent = isNaN(values.platformFeePercent) ? OC_FEE_PERCENT : values.platformFeePercent;
+          const hostFee = Math.round(values.amount * (hostFeePercent / 100));
+          const platformFee = Math.round(values.amount * (platformFeePercent / 100));
+
+          return (
+            <Form>
+              <ModalBody>
+                <P fontSize="16px" lineHeight="24px" fontWeight="500" mt={3}>
+                  <FormattedMessage
+                    id="AddFundsToCollectiveModal.Title"
+                    defaultMessage="Add funds to the Collective:"
```

Comment:
Should we place `:` outside the message? Or including this type of punctuation a good intl practice?

Reply:
Good question, I have no strong opinion on that. Ideally, we want to internationalize that because some languages have a space before the colon, others not. That's why we have https://github.com/opencollective/opencollective-frontend/blob/983d7eb3a0d7b7458955823992a25ae4f119d898/lang/en.json#L1687

No including it in the template message makes it more re-usable. The alternative to write something both re-usable and i18n aware is something like this, a bit verbose but it works:

```jsx
<FormattedMessage
  id="withColon"
  defaultMessage="{item}:"
  values={{ 
    item: <FormattedMessage id="AddFundsToCollectiveModal.Title" defaultMessage="Add funds to the Collective"  />
  }}
/>
```

Reply:
I think it's better to keep it as it is and avoid the i18n composition.